File: scalechain-util/src/main/scala/io/scalechain/util/NetUtil.scala
package io.scalechain.util

import java.net.{InetAddress, NetworkInterface}

import scala.collection.mutable.ArrayBuffer

/**
  * Created by kangmo on 7/11/16.
  */
object NetUtil {
  def getLocalAddresses() : List[String] = {
    val addresses = new ArrayBuffer[String]()
    val e = NetworkInterface.getNetworkInterfaces()
    while(e.hasMoreElements())
    {
      val n : NetworkInterface = e.nextElement()
      val ee = n.getInetAddresses()
      while (ee.hasMoreElements())
      {
        val i : InetAddress = ee.nextElement()
        addresses.append(i.getHostAddress())
      }
    }
    addresses.toList
  }
}

File: scalechain-util/src/main/kotlin/io/scalechain/util/NetUtil.kt
package io.scalechain.util

import java.net.*

/**
 * Created by kangmo on 7/11/16.
 */
object NetUtil {
    @JvmStatic
    fun getLocalAddresses() : List<String> {
        val addresses = arrayListOf<String>()

        // better code suggetion by **b0c1** and **Janos Haber**
        for (iface in NetworkInterface.getNetworkInterfaces()) {
            for (address in iface.getInetAddresses()) {
                addresses.add( address.getHostAddress() )
            }
        }
        return addresses.toList()
/*
        // better code suggetion by **b0c1** and **Janos Haber**
        // But this code is Java 7 code. I need to use Java 6.
        val addresses = Collections.list(NetworkInterface.getNetworkInterfaces())
            .flatMap { iface ->
                Collections.list(iface.getInetAddresses()).map{ it.getHostAddress() }
            }
        return addresses
*/
    }
}
